services/QuillLMS/engines/evidence/lib/evidence/evidence/html_tag_remover.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Evidence
+  class HTMLTagRemover < ApplicationService
+    HTML_TAG_REGEX = /<("[^"]*"|'[^']*'|[^'">])*>/
+
+    attr_reader :html
+
+    def initialize(html)
+      @html = html
+    end
+
+    def run
+      html
+        .gsub(HTML_TAG_REGEX, " ") # remove html tags
```

Comment:
For some HTML structures*, regexes will actually not work for stripping tags.  Fortunately Rails has a [strip_tags](https://api.rubyonrails.org/classes/ActionView/Helpers/SanitizeHelper.html#method-i-strip_tags) method.

*HTML is a context free (type 2) grammar whereas regexes are a regular (type 3) grammar.

Reply:
Oh, good call, going to use `strip_tags` instead.

Reply:
Actually, scratch that, I'll need to improve on this in a future PR. `strip_tags` replaces with a `blank` and we have a lot of stretches with no spaces in our passages, e.g. `Living</strong></p><p><mark>Electricity`, which results in `LivingElectricity` being pulled as a word. I don't have a good way to reliably pull those apart.

Reply:
[loofah](https://github.com/flavorjones/loofah) `to_text` is probably close to what you want:

```
doc = Loofah.fragment("<h1>Title</h1><div>Content<br>Next line</div>")
doc.text    # => "TitleContentNext line"            # probably not what you want
doc.to_text # => "\nTitle\n\nContent\nNext line\n"  # better
```

Then you can just gsub("\n")